cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureTemplateBuilderTest.java

```diff
@@ -567,6 +567,52 @@ public void buildWithGatewayInstanceGroupTypeAndMultipleLoadBalancers(String tem
             "\"id\": \"[resourceId('Microsoft.Network/publicIPAddresses', 'LoadBalancertestStackPUBLIC-publicIp')]\""));
     }
 
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("templatesPathDataProvider")
+    public void testNicDependenciesAreValidJson(String templatePath) {
```

Comment:
could you pls cover the "new network" scenario with tests, too?

Reply:
Yes, it looks like there was a bug there, I added the test `testNicDependenciesWhenNoSecurityGroupsNoFirewallRulesAndNoPublicIp` and got a failure.

I've changed the virtual network dependency from:
```
<#if !noFirewallRules || !noPublicIp>,</#if>
"[concat('Microsoft.Network/virtualNetworks/', parameters('virtualNetworkNamePrefix'))]"
</#if>
```
to:
```
<#if !existingVPC>
   <#if !noPublicIp || (!noFirewallRules && (!isUpscale && (! securityGroups[instance.groupName]?? || ! securityGroups[instance.groupName]?has_content)))>
       ,
  </#if>
   "[concat('Microsoft.Network/virtualNetworks/', parameters('virtualNetworkNamePrefix'))]"
</#if>
```